plugins/inputs/chrony/chrony.go

```diff
@@ -316,22 +317,32 @@ func (c *Chrony) gatherSources(acc telegraf.Accumulator) error {
 		}
 
 		// Trying to resolve the source name
-		sourceNameReq := fbchrony.NewNTPSourceNamePacket(sourceData.IPAddr)
-		sourceNameRaw, err := c.client.Communicate(sourceNameReq)
-		if err != nil {
-			return fmt.Errorf("querying name of source %d failed: %w", idx, err)
-		}
-		sourceName, ok := sourceNameRaw.(*fbchrony.ReplyNTPSourceName)
-		if !ok {
-			return fmt.Errorf("got unexpected response type %T while waiting for source name", sourceNameRaw)
-		}
-
-		// Cut the string at null termination
 		var peer string
-		if termidx := bytes.Index(sourceName.Name[:], []byte{0}); termidx >= 0 {
-			peer = string(sourceName.Name[:termidx])
+
+		if sourceData.Mode == fbchrony.SourceModeRef && sourceData.IPAddr.To4() != nil {
+			// If source is a reference (PPS, etc..), and if the IP is an IPv4 address,
+			// then the name is encoded in its IP (32 bits). Yes
+			//https://gitlab.com/chrony/chrony/-/blob/05bd4898a9a64fe1e5432a78939c4b4b82619c2f/client.c?page=3#L2046
```

Comment:
Maybe
```suggestion
			// References of local sources (PPS, etc..) are encoded in the bits of the IPv4 address,
			// instead of the RefId. Extract the correct name for those sources.
```

Reply:
Hi @srebhan  thanks for your feedback!  Done; please tell me what you think :-)